src/app/my-tab/my-tab.component.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  Component,
+  OnInit,
+  OnDestroy
+} from '@angular/core';
+
+import {
+  AddinClientService
+} from '@blackbaud/skyux-lib-addin-client';
+
+import {
+  AddinClientInitArgs
+} from '@blackbaud/sky-addin-client';
+
+import {
+  Subject
+} from 'rxjs';
+
+import {
+  takeUntil,
+  finalize
+} from 'rxjs/operators';
+
+import {
+  GiftFormData
+} from '../shared/interfaces/gift-form-data';
+
+import {
+  GiftSaveData
+} from '../shared/interfaces/gift-save-data';
+
+import {
+  EntryFormService
+} from '../shared/services/entry-form.service';
+
+@Component({
+  selector: 'my-tab',
+  templateUrl: './my-tab.component.html',
+  styleUrls: ['./my-tab.component.scss']
+})
+export class MyTabComponent implements OnInit, OnDestroy {
+  public environmentId: string;
+  public formData: GiftFormData;
+  public saveData: GiftSaveData;
+  private destroy: Subject<any>;
+
+  constructor(
+    private addinClientService: AddinClientService,
+    private entryFormService: EntryFormService
+  ) {
+    this.destroy = new Subject();
+  }
+
+  public ngOnDestroy() {
+    this.destroy.next();
+    this.destroy.complete();
+  }
+
+  public ngOnInit() {
+    // initialization of the add in client
+    this.initializeAddInClient();
+
+    // capture the update events triggered from the gift form through the add in client
+    this.captureGiftFormUpdateEvent();
+
+    // capture the cancel event triggered from the gift form through the add in client
+    this.captureGiftFormCancelEvent();
+
+    // capture the save event triggered from the gift form through the add in client
+    this.captureGiftFormSaveEvent();
+  }
+
+  private initializeAddInClient(): void {
+    // subscribe to the the add in args
+    this.addinClientService.args.pipe(
+      takeUntil(this.destroy)
+    ).subscribe((args: AddinClientInitArgs) => {
+      this.environmentId = args.envId;
+
+      // initialize the tab
+      args.ready({
+        showUI: true,
+        title: `My entry form Add-in tab`
```

Comment:
May want to shorten this name of the tab. Even just taking out "Add-in" so it becomes "Entry form demo" or something like that. Don't worry about this if you've already started creating screenshots for the doc.

Reply:
Making this change now

Reply:
Renamed add-in tab to "Entry form demo"